README.md

```diff
@@ -1,37 +1,152 @@
-# managed-gitops
+# Managed GitOps Service Monorepo
+
+**This is currently a work in progress and is subject to changes. Do not use this in any production environment.**
 
 ## Overview
 
-For an introduction to the project, and project details, see [Google Drive](https://drive.google.com/drive/u/0/folders/1p_yIOJ1WLu-lqz-BVDn076l1K1pEOc1d).
+> For an introduction to the project, and project details, see [Google Drive](https://drive.google.com/drive/u/0/folders/1p_yIOJ1WLu-lqz-BVDn076l1K1pEOc1d).
+
+This repo is the home for all the Managed GitOps Service components.
+
+This repo contains various tools and services that you can use to deploy, as well as the libraries you can use to develop the GitOps Service that will be integrated with RedHat AppStudio.
+
+---
+
+## Component Index
+
+There are 4 separated, tighly-coupled components:
+
+- [Backend]: responsible for receiving the latest user's intent and then informing both the database and the [Cluster-Agent] about it. It also includes a REST APIServer for webhooks, called [routes].
+- [Cluster-Agent]: responsible for keeping synchronised the [ArgoCD Application CR] and its relevant Application entry in the database.
+- [Backend Shared]: group of libraries shared among the rest of the components.
+- [Frontend]: it is using [PatternFly](https://www.patternfly.org/)/React-based UI, based on PatternFly React Seed.
+
```

Comment:
@jgwest, should we add utilities directory as one more component here as it has relevant load test written?

Reply:
@samyak-jn  +1